src/net_processing.cpp

```diff
@@ -840,7 +875,12 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
     NodeId nodeid = pnode->GetId();
     {
         LOCK(cs_main);
-        mapNodeState.emplace_hint(mapNodeState.end(), std::piecewise_construct, std::forward_as_tuple(nodeid), std::forward_as_tuple(addr, std::move(addrName), pnode->fInbound, pnode->m_manual_connection));
+        mapNodeState.emplace_hint(mapNodeState.end(), std::piecewise_construct, std::forward_as_tuple(nodeid), std::forward_as_tuple(addr, pnode->fInbound, pnode->m_manual_connection));
+    }
+    {
+        PeerRef peer = std::make_shared<Peer>(nodeid);
+        LOCK(g_peer_mutex);
+        g_peer_map.insert(std::make_pair(nodeid, std::move(peer)));
```

Comment:
337528b not sure here, `s/insert/emplace/` to construct rather than copy?

Reply:
ok, I've changed this to `emplace_hint` since the new peers are always added to the end of the map (same as for CNodeStates in mapNodeState)

Reply:
Interesting, TIL thanks, and there were only two `emplace_hint` statements in the codebase, one of which is in the same function. From https://en.cppreference.com/w/cpp/container/map/emplace_hint it looks like a good choice for lower complexity and better perf in this case.